14_contains/solution/contains-solution.spec.js

```diff
@@ -0,0 +1,55 @@
+const contains = require("./contains-solution");
+
+describe("contains", () => {
+  const object = {
+    data: {
+      duplicate: "e",
+      stuff: {
+        thing: {
+          banana: NaN,
+          moreStuff: {
+            something: "foo",
+            answer: [42],
+          },
+        },
+      },
+      info: {
+        duplicate: "e",
+        magicNumber: 44,
+        empty: null,
+      },
+    },
+  };
+
+  test("true if the provided number is a value within the object", () => {
+    expect(contains(object, 44)).toBe(true);
+  });
+
+  test("true if the provided string is a value within the object", () => {
+    expect(contains(object, "foo")).toBe(true);
+  });
+
+  test("does not convert input string into a number when searching for a value within the object", () => {
+    expect(contains(object, "44")).toBe(false);
+  });
+
+  test("false if the provided string is not a value within the object", () => {
+    expect(contains(object, "bar")).toBe(false);
+  });
+
+  test("true if provided string is within the object, even if duplicated", () => {
+    expect(contains(object, "e")).toBe(true);
+  });
+
+  test("false if the provided value exists but is not a primitive", () => {
+    expect(contains(object, [42])).toBe(false);
+  });
```

Comment:
I think we shouldn't make this exclusion for "not a primitive".
This test passes (with `false`) because `[42] === [42]` is false. If you passed in a reference to the same object as one in the nested object, it'd return `true` (no modifications to the current solution).

So we should modify the tests so instead of `[42]`, we define it separately and have `answer: meaningOfLifeArray` or something. Then, we have a test for `contains(object, meaningOfLifeArray)` and `contains(object, [42])` respectively.

Reply:
Ok, I added this